Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3076,6 +3079,11 @@ def visit_AttributeNode(self, node):
         return self.visit_cython_attribute(node)
 
     def visit_NameNode(self, node):
+        if node.name == u'__class__':
+            lenv = self.current_env()
+            entry = lenv.lookup_here(u'__class__')
+            if not entry:
+                node = self._inject_class(node)
         return self.visit_cython_attribute(node)
```

Comment:
`TransformBuiltinMethods` also deals with `locals()/vars()/super()/eval()`. I realise that `__class__` isn't a *method* here but it certainly seems similar to some of those (especially `super()`) which is why it ended up here.

For now I'll update the docstring for `TransformBuiltinMethods` since it seems to have acquired this secondary, undocumented purpose.